src/kOS/Execution/CPU.cs

```diff
@@ -936,20 +942,30 @@ private bool ExecuteInstruction(ProgramContext context)
             Opcode opcode = context.Program[context.InstructionPointer];
             if (DEBUG_EACH_OPCODE)
             {
-                SafeHouse.Logger.Log("ExecuteInstruction.  Opcode number " + context.InstructionPointer + " out of " + context.Program.Count +
-                                      "\n                   Opcode is: " + opcode.Label + " " + opcode.ToString() );
+                executeLog.Append(String.Format("Executing Opcode {0:0000}/{1:0000} {2} {3}\n",
+                                                context.InstructionPointer, context.Program.Count, opcode.Label, opcode.ToString()));
             }
-            
-            if (!(opcode is OpcodeEOF || opcode is OpcodeEOP))
+            try
             {
-                opcode.Execute(this);
-                context.InstructionPointer += opcode.DeltaInstructionPointer;
-                return true;
+                if (!(opcode is OpcodeEOF || opcode is OpcodeEOP))
+                {
+                    opcode.Execute(this);
+                    context.InstructionPointer += opcode.DeltaInstructionPointer;
+                    return true;
+                }
+                if (opcode is OpcodeEOP)
+                {
+                    BreakExecution(false);
+                    SafeHouse.Logger.Log("Execution Broken");
+                }
             }
-            if (opcode is OpcodeEOP)
+            catch (Exception e)
             {
-                BreakExecution(false);
-                SafeHouse.Logger.Log("Execution Broken");
+                // exception will skip the normal printing of the log buffer,
+                // so print what we have so far before throwing up the exception:
+                if (executeLog.Length > 0)
+                    SafeHouse.Logger.Log(executeLog.ToString());
+                throw e;
```

Comment:
if you mean to log and throw i dont think you want 
    
    throw e;

you might instead want 

    throw;

http://theburningmonk.com/2010/02/throwing-exceptions-the-right-way/